### ISSUE TITLE
There is a newer version of tidy-html5 available 

### ISSUE THREAD

**Issue by oalders** [benkasminbullock/html-valid#14]
Thanks for this module!

https://github.com/htacg/tidy-html5/releases/tag/5.8.0 was released after the last release of this module. It's not clear to me me what kind of changes it includes as the release notes are somewhere that is not obvious to me.

**Comment by benkasminbullock** [comment]
Thank you for letting me know about the new version. I had updated it last year but that was before the latest version. I will try to get around to doing the new build soon. Depending on how much has changed, it could be instant or it could take a while. I will update this bug report when it's finished, so there is no need to reply yet.

If you have any other suggestions for improving the module, do let me know.

**Comment by oalders** [comment]
> If you have any other suggestions for improving the module, do let me know.

Thanks! Just for reference, my issue was this:

```perl
#!/usr/bin/env perl

use FindBin '$Bin';
use HTML::Valid;
my $html = <<EOF;
<table><tr><td></tr></table>
EOF
my $htv = HTML::Valid->new();
my ( $output, $errors ) = $htv->run($html);
print "$output\n";
print "$errors\n";
```

The output is:

```
<!DOCTYPE html>
<html>
<head>
<meta name="generator" content=
"HTML Tidy for HTML5 for Apple Unknown OS version 5.0.0">
<title></title>
</head>
<body>
<table>
<tr>
<td></td>
</tr>
</table>
</body>
</html>

line 1 column 1 - Warning: missing <!DOCTYPE> declaration
line 1 column 1 - Warning: inserting implicit <body>
line 1 column 1 - Warning: inserting missing 'title' element
Info: Document content looks like HTML5
Tidy found 3 warnings and 0 errors!
```

`run()` silently adds the missing `</td>` tag, but it's not a warning or error. I didn't see a way to make this an error in the `HTML::Valid` docs. I didn't dig into the `tidy-html5` library, so I may just have a fundamental misunderstanding of how this is supposed to work. I had wanted to use it in a test to ensure that a web app is not emitting broken HTML.

**Comment by oalders** [comment]
> Depending on how much has changed, it could be instant or it could take a while. I will update this bug report when it's finished, so there is no need to reply yet.

To be clear, there is no rush on my end. I just wanted to point out that a new version exists. It's not any kind of a blocker.

**Comment by benkasminbullock** [comment]
The new version should be ready to go soon, I have to do one more round of testing.

I'll have a look at that, but the underlying library of HTML::Valid is a little random, and I had a lot of problems with reporting bugs to them, so I sort of gave up trying to make sense of it. The person who seemed to be the most difficult to deal with on that project was closing my bug reports for reasons like "this bug report makes the page look untidy". He seems to have left the project since last year so I might try again.

The biggest thing this module has to offer is the huge list of HTML tags and attributes, it is directly from the WWW consortium and it's actually very useful. I did discuss with Andy Lester who made HTML::Tagset but I'm not sure what he's doing now with that. Anyway thanks for some encouragement.

**Comment by benkasminbullock** [comment]
Version 5.8.0 of the HTML tidy library is now included in version 0.9 of this module on CPAN.

I've put in an issue about the other thing at the master repository:

https://github.com/htacg/tidy-html5/issues/1043

I'll close this issue since the title job is complete, and maybe we can open another issue about the other thing.

**Comment by oalders** [comment]
> Version 5.8.0 of the HTML tidy library is now included in version 0.9 of this module on CPAN.

Excellent! We will pull this into our dependencies next week.

> I've put in an issue about the other thing at the master repository

👍🏻 I'll be watching that ticket.

Thanks so much for your work on this. It's a valuable tool.
